### PR TITLE
feat(Checkout Leave Modal): Show checkout leave modal for checkout pages.

### PR DESCRIPTION
**Pull request by itsDiwaker**
Show checkout leave modal whenever user tried to cancel the checkout steps by clicking on the
continue shipping button below the order summary section.

#187

### CODE REVIEW

**Comment by itsDiwaker** [comment]
FYI @CristinaSolana:

@cristianPerez I have shown the popup on the click of "Continue Shopping" button(_**going by the description of the popup in functional doc**_) 
Left out the pickGift conditions as commented for now.
Please check if this popup is really needed for Order confirmation page. Though I have added it.

**Few challenges:**
If we go by the description of the ticket above(_**the navigate away scenario**_), we would need to think about a better solution than `window.onbeforeunload`. Because it has below limitations:

- We'll not be able to show our own dialog, as only alert(), confirm() native methods stops the browser navigation when user hits the back/forward button or enters another url in browser address bar or attempts to close the browser tab.

- Event the native alert will show different text on different browsers, we can not programmatically control that.

**Comment by CristinaSolana** [comment]
@itsDiwaker Curious about why "Left out the pickGift conditions as commented for now."? 

Including "Order confirmation" on the ticket was my mistake (please remove). Good catch.

**Comment by itsDiwaker** [comment]
@CristinaSolana 

**Two reasons:**

- I wasn't sure initially whether this popup should only appear when pickGift is enabled. It appeared to me that it's a global warning that we want to show every time a user tries to leave checkout in middle. So haven't spent time on testing that.

- Also I had my doubts about the requirements( _**Navigate away vs Continue Shopping**_) approach.

That's why I haven't spent much time above two points.

Though to first discuss with you these two points before working on them.

File: templates/cart.liquid [hunk 1]
```diff
@@ -304,10 +304,10 @@
                 {% endif %}
                 <a href="/checkout/shipping" class="button secondary expanded large">Checkout</a>
                 {% if websiteSettings.pickGift.enabled != true %}
-                <a href="/shop" class="link">CONTINUE SHOPPING</a>
+                <a href="/shop" class="link js-link-contiue-shopping">CONTINUE SHOPPING</a>
```
Comment (CristinaSolana) [inline templates/cart.liquid]: incorrect spelling of continue in class name, issue present in all other places that use this class.

File: templates/cart.liquid [hunk 2]
```diff
@@ -445,9 +445,13 @@
 </div>
 {% endif %}
 
+{% comment %}
```
Comment (CristinaSolana) [inline templates/cart.liquid]: Why is this commented out? Also commented out in other sections.

**Comment by itsDiwaker** [comment]
@CristinaSolana requested changes have been pushed.

**Pull request closed without merging**